diff --git a/dist/index.mjs b/dist/index.mjs
index 723ce75155576ab5940e9eb843d16966b717a7a3..f7a15c1b7df92ab798cd26cf27d7aa1e1ecb7d50 100644
--- a/dist/index.mjs
+++ b/dist/index.mjs
@@ -408,7 +408,7 @@ const verifyPasskeyRegistration = (options) => {
 					user
 				};
 			};
-			const registration = ctx.body.createSession ? await runWithTransaction(ctx.context.adapter, persistRegistration) : await persistRegistration();
+			const registration = await runWithTransaction(ctx.context.adapter, persistRegistration);
 			if (registration.session && registration.user) {
 				await setSessionCookie(ctx, {
 					session: registration.session,
